src/ErrorHandling/SpectreExceptions.hpp

```diff
@@ -0,0 +1,17 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines custom exception types indicating algorithm failure
```

Comment:
What if we want an exception that isn't an algorithm failure? Where would we put that? I'd just leave this comment out.